deltacat/compute/compactor/compaction_session.py

```diff
@@ -406,8 +462,28 @@ def _execute_compaction_round(
     logger.info(f"Getting {len(dd_tasks_pending)} dedupe results...")
     dd_results: List[DedupeResult] = ray.get(dd_tasks_pending)
     logger.info(f"Got {len(dd_results)} dedupe results.")
+
+    dedupe_end = time.time()
+    compaction_audit.set_dedupe_time_in_seconds(dedupe_end - dedupe_start)
+
+    cluster_utilization_after_dd, cluster_util_after_dd_latency = timed_invocation(
+        ClusterUtilization.get_current_cluster_utilization
+    )
+
+    compaction_audit.set_object_store_memory_used_bytes_by_dedupe(
+        cluster_utilization_after_dd.used_object_store_memory_bytes
+        - cluster_utilization_after_hb.used_object_store_memory_bytes
+    )
+
```

Comment:
Since we're dedicating an increasingly large number of lines to collecting telemetry info, it may be helpful to put larger blocks like this (plus the below lines around peak memory usage and telemetry collection time) inside of a dedicated method like `compaction_audit.save_step_stats("dedupe", dedupe_end - dedupe_start, ddr, cluster_util_after_dd_latency, telemetry_time_dd)`